rust/otap-dataflow/crates/otlp/README.md

```diff
@@ -1,6 +1,40 @@
 # OTLP Nodes
 
-Status: **Skeleton**
+## GenericBatchProcessor
 
-This crate will contain the implementation of the OTLP receiver, OTLP exporter,
-and potentially other OTLP-related processors.
+A flexible, type-safe batch processor for OpenTelemetry Protocol (OTLP) data that leverages Rust's type system for efficient and safe batching of telemetry data.
+
+### Key Features
+
+- **Generic Data Handling**: Uses Rust's type system to handle all OTLP data types (traces, metrics, logs) with a single, unified implementation
+- **Type-Safe Batching**: Ensures type safety at compile time while maintaining efficient batching operations
+- **Preservation of Data Structure**: Maintains the original structure of metrics (Gauges, Sums, Histograms) during batching
+
+### Configurable Flush Triggers
+
+- **Batch Size**: Flushes when the number of items reaches a configurable threshold
+- **Timeout**: Flushes after a specified duration to ensure timely processing
+- **Shutdown**: Guarantees all buffered data is emitted during graceful shutdown
```

Comment:
The collector has a configurable "Sizer" choice which can be "requests", "items" or "bytes".